conda_smithy/templates/travis.yml.tmpl

```diff
@@ -11,6 +11,13 @@ language: generic
 os: osx
 osx_image: xcode6.4
 
+{%- if 'ccache-toolchain' in ' '.join(package['meta']['requirements']['build']) %}
+cache:
+  branch: no-md5deep
```

Comment:
This branch should not be needed anymore: https://github.com/travis-ci/travis-ci/issues/7456#issuecomment-304958133
However I didn't test it after they fixed it. After you make the changes I can help with testing before merging this.